packages/shared/wallet/src/hooks/use-connect.ts

```diff
@@ -34,7 +35,10 @@ export const useConnect = () => {
     (walletLabel?: string) => {
       // No change to wallets, nothing to do here.
       if (!walletLabel || currentLabel === walletLabel) return;
-      connect({ autoSelect: { label: walletLabel, disableModals: true } });
+      connect({ autoSelect: { label: walletLabel, disableModals: true } }).then((resp) => {
+        // NOTE: this handles initial wallet connection
```

Comment:
This will fire multiple times unless it goes into the observables. Same goes for the useEffect.

https://github.com/notional-finance/notional-monorepo/blob/be08117fb2910814092edc1abc8852a384296e60/packages/notionable/src/global/sagas/on-wallet-connect.ts

These should get fired off here:
![image](https://github.com/notional-finance/notional-monorepo/assets/977434/053a207d-bbb7-47c0-a4c7-dbdde67f7ab5)